packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -804,7 +825,11 @@ export class ConnectionManager implements IConnectionManager {
 			0x0e8 /* "readonly perf with write connection" */,
 		);
 
-		this.set_readonlyPermissions(readonly, oldReadonlyValue);
+		this.set_readonlyPermissions(
+			readonly,
+			oldReadonlyValue,
+			isNoDeltaStreamConnection(connection) ? connection.error?.errorType : undefined,
```

Comment:
i would also use IConnectionStateChangeReason in NoDeltaStreamConnection, se we track and propagate both error and reason